Add API to get all scheduled uploads

hojason117 opened the pull request:
# Summary

Add `getAllUploads` API to retrieve all scheduled uploads that are in the system's internal database along with their status.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅   |
| Android |    ✅   |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I updated the typed files (TS)
- [ ] I've added Detox End-to-End Test(s)
- [ ] I've created a snack to demonstrate the changes: LINK HERE

hojason117 committed:
Add `getAllUploads` method

hojason117 committed:
Use UUID for uploadId on iOS

The previous implementation uses an integer starting at 0 on every app launch. This guarantees
ID collisions across app sessions and thus makes it impossible to track uploads scheduled
during previous sessions unless setting `customUploadId`.

hojason117 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add an API to retrieve all scheduled upload](https://trello.com/c/KkNFaNJ4/164-add-an-api-to-retrieve-all-scheduled-upload)

Karthz commented on android/src/main/java/com/appfolio/uploader/UploaderModule.kt:
```diff
@@ -312,6 +315,38 @@ class UploaderModule(val reactContext: ReactApplicationContext) : ReactContextBa
+  @ReactMethod
+  fun getAllUploads(promise: Promise) {
+    val workManager: WorkManager = WorkManager.getInstance(reactContext)
+    val works = workManager.getWorkInfosByTag(UploadWorker::class.java.name).get()
```
nit: works --> workInfos

Karthz commented on ios/VydiaRNFileUploader.m:
```diff
@@ -273,6 +270,39 @@ - (void)copyAssetToFile: (NSString *)assetUrl completionHandler: (void(^)(NSStri
+        NSMutableArray *uploads = [NSMutableArray new];
+        for (NSURLSessionUploadTask *uploadTask in uploadTasks) {
+            NSString *state;
+            switch(uploadTask.state){
```
nit: space before {

hojason117 committed:
Reduce NSURLSession initialization delay to 5 seconds

We need enough delay in order to give JS time to add its listeners but
not too long that we start wasting background credit.

Karthz commented:
:tada: This PR is included in version 6.9.0 :tada:

The release is available on:
- `npm package (@latest dist-tag)`
- [GitHub release](https://github.com/appfolio/react-native-upload/releases/tag/v6.9.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
